cli/azd/pkg/apphost/generate.go

```diff
@@ -828,14 +828,16 @@ func (b *infraGenerator) buildEnvBlock(env map[string]string, manifestCtx *genCo
 		// And YAML rules would treat the above as the value being a boolean instead of a string, which the container
 		// app service expects.
 		//
-		// JSON marshalling the string value will give us something like `"true"` (with the quotes, and any escaping
-		// that needs to be done), which is what we want here.
-		jsonStr, err := json.Marshal(res)
+		// YAML marshalling the string value will give us something like `"true"` (with the quotes, and any escaping
+		// that needs to be done), which is what we want here. The TrimSpace is to remove any leading or trailing whitespace
+		// in the string, as yaml.Marshal("true") will give back a string with a trailing newline.
+		yamlStr, err := yaml.Marshal(res)
```

Comment:
I do wonder if we should remove the responsibility of this yaml formatting out of this layer and into the templates.  I know in helm (which also uses text templates to produce yaml manifests) there are helpers that you can call in the go text template to get yaml values like this.  That might make things easier long term.  Something for us to think about longer term, I think this is the most direct fix we can make right now.

Reply:
(feel free to resolve this when you merge this, @vhvb1989, no action needs to be taken, just wanted to give the thought in some context).